src/atomicassets.cpp

```diff
@@ -503,6 +503,46 @@ ACTION atomicassets::locktemplate(
 }
 
 
+/**
+* Reduces the max supply of the template to the new max supply
+* This means that afterwards, NFTs can only be minted up to the new max supply
+* @required_auth authorized_editor, who is within the authorized_accounts list of the collection
+**/
+
+ACTION atomicassets::redmaxtemplt(
+    name authorized_editor,
+    name collection_name,
+    int32_t template_id,
+    uint32_t new_max_supply
+) {
+    require_auth(authorized_editor);
+
+    auto collection_itr = collections.require_find(collection_name.value,
+        "No collection with this name exists");
+
+    check_has_collection_auth(
+        authorized_editor,
+        collection_name,
+        "The editor is not authorized within the collection"
+    );
+
+    templates_t collection_templates = get_templates(collection_name);
+    auto template_itr = collection_templates.require_find(template_id,
+        "No template with the specified id exists for the specified collection");
+
+    check(template_itr->issued_supply != 0,
+        "Can't lock a template that does not have at least one issued asset");
+
+    check(new_max_supply > 0, "Can't set the max supply of a template to 0");
+
+    check(template_itr->max_supply > new_max_supply,"Can't increase the max supply of a template");
```

Comment:
This check needs to also consider the case where the current max supply is 0 (unlimited)